crates/ruff_python_formatter/src/parentheses.rs

```diff
@@ -68,7 +68,9 @@ impl<'a> Visitor<'a> for ParenthesesNormalizer {
             }
             StmtKind::For { target, iter, .. } | StmtKind::AsyncFor { target, iter, .. } => {
                 use_inferred_parens(target);
-                use_inferred_parens(iter);
+                if !matches!(iter.node, ExprKind::Tuple { .. }) {
```

Comment:
by using https://github.com/charliermarsh/ruff/pull/3159#discussion_r1115198507,

it can be 

```suggestion
                if !iter.node.is_tuple() {
```
without manually adding methods